### PR TITLE
Add option to hide server messages in chat

### PR DESCRIPTION
**Pull request by ChillerDragon**
cl_filterchat 2 stays unchanged and still shows server messages
cl_filterchat 3 deactivates all chat messages

Can come in handy for cinematic videos or the pure silence :)

### CODE REVIEW

**Comment by Dune-jr** [comment]
Have you tested the changes to the settings UI? I'm afraid the new long string overflows

**Comment by ChillerDragon** [comment]
screenshot_2019-11-19_20-00-35

@Dune-jr yea thought the same but using ``+`` instead of ``and`` shortend it :)

File: src/game/client/components/menus_settings.cpp [hunk 1]
```diff
@@ -1065,12 +1065,14 @@ void CMenus::RenderSettingsGeneral(CUIRect MainView)
 		if(g_Config.m_ClFilterchat == 0)
 			str_format(aBuf, sizeof(aBuf), Localize("everyone", "Show chat messages from:"));
 		else if(g_Config.m_ClFilterchat == 1)
-			str_format(aBuf, sizeof(aBuf), Localize("friends only", "Show chat messages from:"));
+			str_format(aBuf, sizeof(aBuf), Localize("friends and server", "Show chat messages from:"));
```
Comment (Dune-jr) [inline src/game/client/components/menus_settings.cpp]: I am talking about this

**Comment by ChillerDragon** [comment]
screenshot_2019-11-21_10-18-14

Oh yea I tested that as well. There is not too much padding left but it is inside the box. I thought this is a better trade-off than replacing the ``and`` with the ``+`` in the settings ui.

**Comment by Dune-jr** [comment]
Unfortunately, this is too tight for translations, which is why there must be some padding to labels.

**Comment by ChillerDragon** [comment]
@Dune-jr yea translation is a problem but otherwise I am not very happy with the pr. It is sad that messages are lost that are sent during the time the chat is off.

Would be nice to have another option or replace the current one the skip rendering chat but continues to receive messages.

I see the current use case is for tee's who want to focus on the game and only see messages from the server. But another big use case is taking a screenshot or recording a quick cinematic clip and turn the chat off. But afterwards it is sad when the messages are lost.

**Pull request closed without merging**